internal/services/datafactory/data_factory_credential_user_assigned_managed_identity.go

```diff
@@ -0,0 +1,288 @@
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.DataFactory.Credentials
+
+			var data DataFactoryCredentialUserAssignedManagedIdentityResourceSchema
+			if err := metadata.Decode(&data); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			dataFactoryId, err := credentials.ParseFactoryID(data.DataFactoryId)
+			if err != nil {
+				return err
+			}
+
+			id := credentials.CredentialId{
+				SubscriptionId:    dataFactoryId.SubscriptionId,
+				ResourceGroupName: dataFactoryId.ResourceGroupName,
+				FactoryName:       dataFactoryId.FactoryName,
+				CredentialName:    data.Name,
+			}
+
+			existing, err := client.CredentialOperationsGet(ctx, id, credentials.CredentialOperationsGetOperationOptions{})
+			if err != nil {
+				if existing.HttpResponse.Status == "404" {
+					return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+				}
+			}
+			if existing.HttpResponse.Status == "404" {
+				return tf.ImportAsExistsError("azurerm_data_factory_dataset_http", id.ID())
+			}
+
+			credential := credentials.ManagedIdentityCredentialResource{
+				Type: utils.String(IDENTITY_TYPE),
+				Properties: credentials.ManagedIdentityCredential{
+					TypeProperties: &credentials.ManagedIdentityTypeProperties{},
+				},
+			}
+
+			if len(data.Annotations) > 0 {
+				annotations := make([]interface{}, len(data.Annotations))
+				for i, v := range data.Annotations {
+					annotations[i] = v
+				}
+				credential.Properties.Annotations = &annotations
+			}
+
+			if data.Description != "" {
+				credential.Properties.Description = &data.Description
+			}
+
+			if data.IdentityId != "" {
+				credential.Properties.TypeProperties.ResourceId = &data.IdentityId
+			}
+
+			_, err = client.CredentialOperationsCreateOrUpdate(ctx, id, credential, credentials.CredentialOperationsCreateOrUpdateOperationOptions{})
+			if err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+
+			return nil
+		},
+	}
+}
+
+func (r DataFactoryCredentialUserAssignedManagedIdentityResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.DataFactory.Credentials
+			id, err := credentials.ParseCredentialID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var data DataFactoryCredentialUserAssignedManagedIdentityResourceSchema
+			if err := metadata.Decode(&data); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			existing, err := client.CredentialOperationsGet(ctx, *id, credentials.CredentialOperationsGetOperationOptions{})
+			if err != nil {
+				if existing.HttpResponse.Status == "404" {
+					return fmt.Errorf("checking for presence of existing %s: %+v", id.ID(), err)
+				}
+			}
+
+			credential := credentials.ManagedIdentityCredentialResource{
```

Comment:
We can check to see if Model is nil and if it is we error. And then set credential to existing.Model

```suggestion
                         if existing.Model == nil {
                           return fmt.Errorf("model was nil for %s", id)
                         }
			credential := * existing.Model
```